scripts/ci/build-libcurl.sh

```diff
@@ -134,7 +134,9 @@ fi
 #####
 # ssl
 ####
-if [ ! -z "$OPENSSL_BUILD_FOLDER" ]; then
+if [ "${RUNNER_OS}" == "macOS" ]; then
```

Comment:
According to the information from [curl doc](https://curl.se/docs/install.html), `Secure Transport` is the native implementation of TCL on macOS. It can read the keychain on macOS correctly, but `OpenSSL` can **NOT**.

<img width="1052" alt="image" src="https://github.com/Kong/node-libcurl/assets/130019718/6a3181d6-e47a-44f9-9592-210174449d13">